src/cargo/ops/cargo_new.rs

```diff
@@ -412,74 +510,84 @@ fn mk(config: &Config, opts: &MkOptions) -> CargoResult<()> {
         (None, None, name, None) => name,
     };
 
-    let mut cargotoml_path_specifier = String::new();
-
-    // Calculare what [lib] and [[bin]]s do we need to append to Cargo.toml
+    let template_dir = try!(get_input_template(config, opts));
+    let template_path = match opts.template {
+        Some(template) => template_dir.path().join(template),
+        None => template_dir.path().to_path_buf()
+    };
 
-    for i in &opts.source_files {
-        if i.bin {
-            if i.relative_path != "src/main.rs" {
-                cargotoml_path_specifier.push_str(&format!(r#"
-[[bin]]
-name = "{}"
-path = {}
-"#, i.target_name, toml::Value::String(i.relative_path.clone())));
-            }
-        } else {
-            if i.relative_path != "src/lib.rs" {
-                cargotoml_path_specifier.push_str(&format!(r#"
-[lib]
-name = "{}"
-path = {}
-"#, i.target_name, toml::Value::String(i.relative_path.clone())));
-            }
-        }
+    // make sure that the template exists
+    if fs::metadata(&template_path).is_err() {
+        return Err(human(format!("Template `{}` not found", template_path.display())))
     }
 
-    // Create Cargo.toml file with necessary [lib] and [[bin]] sections, if needed
-
-    try!(paths::write(&path.join("Cargo.toml"), format!(
-r#"[package]
-name = "{}"
-version = "0.1.0"
-authors = [{}]
-
-[dependencies]
-{}"#, name, toml::Value::String(author), cargotoml_path_specifier).as_bytes()));
-
-
-    // Create all specified source files
-    // (with respective parent directories)
-    // if they are don't exist
-
-    for i in &opts.source_files {
-        let path_of_source_file = path.join(i.relative_path.clone());
+    // construct the mapping used to populate the template
+    // if in the future we want to make more varaibles available in
+    // the templates, this would be the place to do it.
+    let mut handlebars = Handlebars::new();
+    // We don't want html escaping...
+    handlebars.register_escape_fn(no_escape);
+
+    let mut data = BTreeMap::new();
+    data.insert("name".to_owned(), name.to_owned());
+    let author_value = format!("{}", toml::Value::String(author));
```

Comment:
Is there a way to express a list in handlebars syntax? It seems somewhat odd to me to assume that we always want TOML escaping here?